src/websocket_endpoint.cpp

```diff
@@ -108,5 +138,108 @@ namespace sensr
     }     
   }
 
+  /// Verify that one of the subject alternative names matches the given hostname
+  bool WebSocketEndPoint::VerifySubjectAlternativeName(const char * hostname, X509 * cert) {
+    STACK_OF(GENERAL_NAME) * san_names = NULL;    
+    san_names = (STACK_OF(GENERAL_NAME) *) X509_get_ext_d2i(cert, NID_subject_alt_name, NULL, NULL);
```

Comment:
If I'm understanding this weird code correctly, this should work
```suggestion
    san_names = static_cast<STACK_OF(GENERAL_NAME) *>(X509_get_ext_d2i(cert, NID_subject_alt_name, nullptr, nullptr));
```